pkg/agent/config/config.go

```diff
@@ -630,6 +630,24 @@ func get(ctx context.Context, envInfo *cmds.Agent, proxy proxy.Proxy) (*config.N
 		nodeConfig.AgentConfig.NodeExternalIP = nodeConfig.AgentConfig.NodeExternalIPs[0].String()
 	}
 
+	var nodeExternalDNSs []string
+	for _, dnsString := range envInfo.NodeExternalDNS.Value() {
+		nodeExternalDNSs = append(nodeExternalDNSs, strings.Split(dnsString, ",")...)
+	}
+	nodeConfig.AgentConfig.NodeExternalDNSs = nodeExternalDNSs
+	if len(nodeConfig.AgentConfig.NodeExternalDNSs) > 0 {
+		nodeConfig.AgentConfig.NodeExternalDNS = nodeConfig.AgentConfig.NodeExternalDNSs[0]
+	}
```

Comment:
you don't need to do this. The NodeInternalIP/NodeExternalIP options used to take only a single value, so we had to maintain backwards compatibility for that by passing the first list value into the old single-valued variable.

Get rid of `NodeExternalDNS` and `NodeInternalDNS` entirely.